server/metrics/metrics.go

```diff
@@ -333,57 +342,51 @@ func (m *metrics) ObserveConnectedUsers(count int64) {
 	}
 }
 
-func (m *metrics) ObserveChangeEventTotal(changeType string) {
-	if m != nil {
-		m.changeEventTotal.With(prometheus.Labels{"change_type": changeType}).Inc()
-	}
-}
-
-func (m *metrics) ObserveProcessedChangeEventTotal(changeType string, discardedReason string) {
+func (m *metrics) ObserveChangeEvent(changeType string, discardedReason string) {
 	if m != nil {
-		m.processedChangeEventTotal.With(prometheus.Labels{"change_type": changeType, "discarded_reason": discardedReason}).Inc()
+		m.changeEventsTotal.With(prometheus.Labels{"change_type": changeType, "discarded_reason": discardedReason}).Inc()
 	}
 }
 
-func (m *metrics) ObserveLifecycleEventTotal(lifecycleEventType string) {
+func (m *metrics) ObserveLifecycleEvent(lifecycleEventType string) {
```

Comment:
What's the reason to drop the suffix Count/Total? Is this a recommended practice as well? I am not seeing any mention of that here: https://prometheus.io/docs/practices/naming/

Reply:
For the actual metric, we're keeping the `_total` suffix for counters like this one:
https://github.com/mattermost/mattermost-plugin-msteams-sync/blob/456916c3f744fb6b099696be7c83832580e1b2dc/server/metrics/metrics.go#L187-L190

In naming the API method, I opted for the more correct "ObserveThing" -- i.e. we saw one thing, and we're incrementing our count of that in a variable named `..._total`. In the examples on the linked Prometheus page, if there's an obvious unit, we use that (e.g. `_seconds`). But then for no units it gives:

> `http_requests_total` (for a unit-less accumulating count)

As most of our counters are unitless, I've converged on `_total` in most cases.

Note that for gauges, I inferred from other documentation /not/ to use `..._total` as that was meant to be reserved for counters. So in those cases, I just state the name of the thing, e.g. `synthetic_users` and `linked_channels`.

Reply:
Yeah I was talking about the API method naming. Your reasoning makes sense. :+1: